addons/stock/views/stock_lot_views.xml

```diff
@@ -34,26 +34,28 @@
                     </h1>
                 </div>
                 <group name="main_group">
-                    <group>
+                    <group string="Information">
                         <field name="product_id" context="{'default_is_storable': True, 'default_tracking': 'lot'}" readonly="context.get('set_product_readonly', False)" force_save="1" help="Product this lot/serial number contains. You cannot change it anymore if it has already been moved."/>
+                        <field name="ref" string="Reference"/>
+                        <field name="company_id" groups="base.group_multi_company" placeholder="Visible to all"/>
+                        <field name="partner_ids" string="Customer" invisible="not partner_ids or not location_id" groups="stock.group_stock_manager"
+                               widget="many2many_tags" options="{'edit_tags': True}"/>
+                    </group>
+                    <group string="Inventory" name="inventory_group">
                         <label for="product_qty" invisible="not display_complete"/>
-                        <div class="o_row" invisible="not display_complete">
-                            <field name="product_qty"/>
-                            <field name="product_uom_id" groups="uom.group_uom" widget="many2one_uom"/>
+                        <div class="o_row" invisible="not display_complete" style="width: 10% !important;">
+                            <field name="product_qty" style="width: fit-content !important;"/>
```

Comment:
May be cleaner to wrap it in a div + add `class="oe_inline"` then use raw CSS, but not sure since it looks like we're not consistent. E.g.:

https://github.com/odoo/odoo/blob/75d08621a3f20e9352106eacd5419691a9ca6ed1/addons/product/views/product_views.xml#L73-L91

results in:

<img width="1215" height="421" alt="image" src="https://github.com/user-attachments/assets/63a1a8ae-2c87-4a41-9735-4ddf94f48175" />